gateway-service/src/main/java/org/zowe/apiml/gateway/ribbon/loadbalancer/predicate/AuthenticationBasedPredicate.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package org.zowe.apiml.gateway.ribbon.loadbalancer.predicate;
+
+import com.netflix.niws.loadbalancer.DiscoveryEnabledServer;
+import com.netflix.zuul.context.RequestContext;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.zowe.apiml.gateway.cache.LoadBalancerCache;
+import org.zowe.apiml.gateway.ribbon.loadbalancer.LoadBalancingContext;
+import org.zowe.apiml.gateway.ribbon.loadbalancer.RequestAwarePredicate;
+import org.zowe.apiml.gateway.ribbon.loadbalancer.model.LoadBalancerCacheRecord;
+import org.zowe.apiml.gateway.security.service.HttpAuthenticationService;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Optional;
+
+import static org.springframework.cloud.netflix.zuul.filters.support.FilterConstants.SERVICE_ID_KEY;
+
+/**
+ * Based on the authentication information decide which instance should be used.
+ * If the user is authenticated and already has routing information stored in cache, use the information.
+ * <p>
+ * There is also terrible overhead as this happens for all instance ids.
+ */
+@RequiredArgsConstructor
+public class AuthenticationBasedPredicate extends RequestAwarePredicate {
+    private final HttpAuthenticationService authenticationService;
+    private final LoadBalancerCache cache;
+
+    @Value("${instance.metadata.apiml.lb.cacheRecordExpirationTimeInHours:8}")
```

Comment:
Consider adding a key for this predicate to separate configs for different predicates. I'd recommend something like `apiml.lb.auth.cacheRecordExpirationTimeInHours`. We don't support it now but we might later and this change is very cheap now.